cpp/src/traversal/bfs_kernels.cuh

```diff
@@ -856,12 +863,18 @@ __global__ void topdown_expand_kernel(
           IndexType gid               = block_offset + thread_item_index * blockDim.x + threadIdx.x;
 
           IndexType row_ptr_u = vec_row_ptr_u[iv];
-          IndexType edge      = row_ptr_u + gid - vec_frontier_degrees_exclusive_sum_index[iv];
-
-          if (edge_mask && !edge_mask[edge]) row_ptr_u = -1;  // disabling edge
-
-          // Destination of this edge
-          vec_dest_v[iv] = (row_ptr_u != -1) ? col_ind[edge] : -1;
+          // Need this check so that we don't use invalid values of edge to index
+          if (row_ptr_u != ~IndexType(0)) {
+            IndexType edge = row_ptr_u + gid - vec_frontier_degrees_exclusive_sum_index[iv];
+
+            if (edge_mask && !edge_mask[edge]) {
+              // Disabling edge
+              row_ptr_u = ~IndexType(0);
```

Comment:
invalid_vertex_id<IndexType>::value is much more readable than ~IndexType(0)

Reply:
Done.